playground/src/components/playground/NavigationPanel/RoutesPanel/RoutesPanel.tsx

```diff
@@ -3,10 +3,52 @@ import { useNavigate } from "@tanstack/react-router";
 import { useMemo, useRef, useState } from "react";
 import { useHotkeys } from "react-hotkeys-hook";
 import { useStudioStore } from "../../store";
+import type { ApiRoute } from "../../types";
 import { Search } from "../Search";
 import { RoutesItem } from "./RoutesItem";
 import { RoutesSection } from "./RoutesSection";
 
+function useRoutesGroupedByTags(routes: ApiRoute[]) {
+  const routesGroupedByTags = useMemo(() => {
+    return routes.reduce<Record<string, ApiRoute[]>>((acc, route) => {
+      // HACK - Also group routes without tags under the "untagged" tag
+      const tags = route.tags?.length ? route.tags : ["untagged"];
+
+      // Add route to each of its tags
+      for (const tag of tags) {
+        if (!acc[tag]) {
+          acc[tag] = [];
+        }
+        acc[tag].push(route);
+      }
+
+      return acc;
+    }, {});
+  }, [routes]);
+
+  // Get sorted tag names (TODO: respect tag order from spec instead of doing alphabetical ordering)
+  const sortedTags = useMemo(() => {
+    const tags = Object.keys(routesGroupedByTags);
+    const untagged = "untagged";
```

Comment:
I guess this is fine, but was thinking that perhaps a symbol would have been a better approach

Reply:
no that is better - guarantees no clash with the api, i changed it